x-pack/plugins/apm/ftr_e2e/cypress/e2e/power_user/onboarding/onboarding.cy.ts

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { apm, timerange } from '@kbn/apm-synthtrace-client';
+import { synthtrace } from '../../../../synthtrace';
+
+const start = Date.now() - 1000;
+const end = Date.now();
+
+function generateData({ from, to }: { from: number; to: number }) {
+  const range = timerange(from, to);
+  const synthGo1 = apm
+    .service({
+      name: 'synth-go-1',
+      environment: 'production',
+      agentName: 'go',
+    })
+    .instance('my-instance');
+
+  return range.interval('1m').generator((timestamp) => {
+    return [
+      synthGo1
+        .transaction({ transactionName: 'GET /apple 🍎' })
+        .timestamp(timestamp)
+        .duration(1000)
+        .success(),
+    ];
+  });
+}
+
+describe('APM Onboarding', () => {
+  describe('General navigation', () => {
+    beforeEach(() => {
+      cy.loginAsEditorUser();
+      cy.visitKibana('/app/apm/onboarding');
+    });
+
+    it('includes section for APM Agents', () => {
+      cy.contains('APM Agents');
+      cy.contains('Node.js');
+      cy.contains('Django');
+      cy.contains('Flask');
+      cy.contains('Ruby on Rails');
+      cy.contains('Rack');
+      cy.contains('Go');
+      cy.contains('Java');
+      cy.contains('.NET');
+      cy.contains('PHP');
+      cy.contains('OpenTelemetry');
+    });
+
+    it('navigation to different Tabs', () => {
+      cy.contains('Django').click();
+      cy.contains('pip install elastic-apm');
+
+      cy.contains('Flask').click();
+      cy.contains('pip install elastic-apm[flask]');
+
+      cy.contains('Ruby on Rails').click();
+      cy.contains("gem 'elastic-apm'");
+
+      cy.contains('Rack').click();
+      cy.contains("gem 'elastic-apm'");
+
+      cy.contains('Go').click();
+      cy.contains('go get go.elastic.co/apm');
+
+      cy.contains('Java').click();
+      cy.contains('-javaagent');
+
+      cy.contains('.NET').click();
+      cy.contains('Elastic.Apm.NetCoreAll');
+
+      cy.contains('PHP').click();
+      cy.contains('apk add --allow-untrusted <package-file>.apk');
+
+      cy.contains('OpenTelemetry').click();
+      cy.contains('Download the OpenTelemetry APM Agent or SDK');
+    });
+  });
+
+  describe('check Agent Status', () => {
+    beforeEach(() => {
+      cy.loginAsEditorUser();
+      cy.visitKibana('/app/apm/onboarding');
+    });
+    it('when no data is present', () => {
+      cy.intercept('GET', '/internal/apm/agent_status').as('agentStatus');
+      cy.getByTestSubj('checkAgentStatus').click();
+      cy.wait('@agentStatus');
+      cy.getByTestSubj('agentStatusWarningCallout');
```

Comment:
Shouldn't this be an expectation of some sorts? 

```suggestion
      cy.getByTestSubj('agentStatusWarningCallout').should('exist');
```